numpy/lib/function_base.py

```diff
@@ -52,6 +53,187 @@
+        q: the quantile
+        n: the size of the studied sample
+        i: lower bound
+        j: upper bound
+    There are 9 methods listed in the literature (see Hyndman&Fan below):
+    For continuous methods (4, 5, 6, 7, 8, 9), the interpolation can be
+    described by : (q - alpha) / ( n - alpha - beta + 1 )
+
+    * INVERTED_CDF:
+        method 1 of H&F.
+        This method give discontinuous results:
+            if g > 0 ; then take j
+            if g = 0 ; then take i
+    * AVERAGED_INVERTED_CDF:
+        method 2 of H&F.
+        This method give discontinuous results:
+            if g > 0 ; then take j
+            if g = 0 ; then average between bounds
+    * CLOSEST_OBSERVATION:
+        method 3 of H&F.
+        This method give discontinuous results:
+            if g > 0 ; then take j
+            if g = 0 and index is odd ; then take j
+            if g = 0 and index is even ; then take i
+    * INTERPOLATED_INVERTED_CDF:
+        method 4 of H&F.
+        This method give continuous results using:
+            alpha = 0
+            beta = 1
+    * HAZEN:
+        method 5 of H&F.
+        This method give continuous results using:
+            alpha = 1/2
+            beta = 1/2
+    * WEIBULL:
+        method 6 of H&F.
+        This method give continuous results using:
+            alpha = 0
+            beta = 0
+    * INCLUSIVE:
+        method 7 of H&F.
+        It the default method for historical reasons.
+        Using "linear" will alias to this method.
+        This method give continuous results using:
+            alpha = 1
+            beta = 1
+    * MEDIAN_UNBIASED:
+        method 8 of H&F.
+        This method is probably the best method if the sample distribution
+        function is unknown.
+        This method give continuous results using:
+            alpha = 1/3
+            beta = 1/3
+    * NORMAL_UNBIASED:
+        method 9 of H&F.
+        This method is probably the best method if the sample distribution
+        function is known to be normal.
+        This method give continuous results using:
+            alpha = 3/8
+            beta = 3/8
+    Additionally, numpy has historically provided 4 others methods.
+    Beware, these 4 methods are not described in the scientific literature.
+    * 'lower': ``i``.
+    * 'higher': ``j``.
+    * 'nearest': ``i`` or ``j``, whichever is nearest.
+    * 'midpoint': ``(i + j) / 2``.
+
+    References
+    ----------
+    [Hyndman&Fan] Hyndman, R. J., & Fan, Y. (1996).
+    Sample quantiles in statistical packages.
+    The American Statistician, 50(4), 361-365.
+    """
+    # --- HYNDMAN and FAN methods
+    INVERTED_CDF = (
+        {"inverted_cdf"},
+        lambda n, quantiles: (n * quantiles) - 1,
+        lambda gamma, _: _get_gamma_mask(shape=gamma.shape,
+                                         default_value=1.,
+                                         value=0.,
+                                         where=gamma == 0),
```

Comment:
Should this be implemented as a correction to the first formula?  So that `gamma` can just be zeros (or even a special value `None` for non-interpolating variations?

Reply:
The same goes for `closest_observation`, which is also a discrete method (without any interpolation).

Reply:
I think it can be part of the first formula, but it is again to match what is described by H&F paper and ease understanding of this code.
Is there a benefit to integrate it in the first formula ?

Reply:
The reason I suggested it was as a way to skip the linear interpolation step for the discrete methods.  Unless that is the easiest thing to achieve that, there is no point.

Reply:
it's done.